pyccl/tracers.py

```diff
@@ -262,6 +321,95 @@ def get_bessel_derivative(self):
+        if ((bias_transfer_a is None) and (bias_transfer_k is None)):    
+            if (isinstance(k_MG, (list, np.ndarray))): 		
+                # Scale-dependent case with full k array 
+                self.add_tracer(cosmo, kernel,
+                    transfer_ka=mg_transfer, der_bessel=der_bessel,
+                    der_angles=der_angles)
+            elif (isinstance(k_MG, float) or isinstance(k_MG, int)): 		
+                # Scale-dependent case with  a single specified k value
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer, der_bessel=der_bessel,
+                    der_angles=der_angles)
+            else:		
+                # Scale-independent case where k=None
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer, der_bessel=der_bessel, 
+                    der_angles=der_angles)
+
+        #  case of an astro bias depending on a and  k 
+        elif ((bias_transfer_a is not None) and (bias_transfer_k is not None)):   
+            if (isinstance(k_MG, (list, np.ndarray))):
+                # Scale-dependent case with full k array 
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                (bias_transfer_a[1] * (bias_transfer_k[1]*mg_transfer[2]).T).T)
+                self.add_tracer(cosmo, kernel,
+                    transfer_ka=mg_transfer_new,
+                    der_bessel=der_bessel, der_angles=der_angles)	
+            elif (isinstance(k_MG, float) or isinstance(k_MG, int)):
+                # Scale-dependent case with  a single specified k value
+                mg_transfer_new = (mg_transfer[0] ,mg_transfer[1] * bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer_new,
+                    der_bessel=der_bessel, der_angles=der_angles, transfer_k=bias_transfer_k )
+            else:
+                # Scale-independent case where k=None
+                mg_transfer_new = (mg_transfer[0] ,mg_transfer[1] * bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer_new,
+                    der_bessel=der_bessel, der_angles=der_angles, transfer_k=bias_transfer_k )
+				
+        #  case of an astro bias depending on a but not k  
+        elif ((bias_transfer_a is not None) and (bias_transfer_k is None)):    
+            if (isinstance(k_MG, (list, np.ndarray))):
+                # Scale-dependent case with full k array 
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1], 
+                                (bias_transfer_a[1] * mg_transfer[2].T).T)
+                self.add_tracer(cosmo, kernel,
+                    transfer_ka=mg_transfer_new,
+                     der_bessel=der_bessel, der_angles=der_angles)
+            elif (isinstance(k_MG, float) or isinstance(k_MG, int)):
+                # Scale-dependent case with  a single specified k value
```

Comment:
second two cases look the same again here

Reply:
see response above.